blog/2025-06-15-temporal-impl-1.md

```diff
@@ -0,0 +1,439 @@
+
+// FiniteF64 usage
+let number: f64 = js_value.to_number(context)?;
+let finite_f64: FiniteF64 = FiniteF64::try_from(number)?;
+let year: i32 = finite_f64.as_integer_with_truncation::<i32>();
+
+// FromStr usage with `get_option`
+let options_obj: &JsObject = get_options_object(&js_value)?;
+let overflow: Option<ArithmeticOverflow> = get_option::<ArithmeticOverflow>(
+    &options_obj,
+    js_string!("overflow"),
+    context
+)?;
+```
+
+This is the core glue between Boa and the `temporal_rs` API that we will
+be going over below.
+
+## Implementing constructors
+
+There are a variety of ways to construct a core component like
+`PlainDate`, and that stems from the core constructor for each of the
+core components `new_with_overflow`:
+
+```rust
+impl PlainDate {
+    pub fn new_with_overflow(year: i32, month: u8, day: u8, calendar: Calendar, overflow: ArithmeticOverflow) -> Result<Self> {
+        // Create PlainDate
+    }
+}
+```
+
+This function supports the baseline construction of Temporal builtins,
+which takes the usual year, month, day, alongside a calendar and also an
+overflow option to constrain or reject based on whether the provided
+values are in an expected range.
+
+However, we can better express this in Rust with common `try_` prefix
+notation.
+
+```rust
+impl PlainDate {
+    pub fn new(year: i32, month: u8, day: u8, calendar: Calendar) -> Result<Self> {
+        Self::new_with_overflow(year, month, day, calendar, ArithmeticOverflow::Constrain)
+    }
+
+    pub fn try_new(year: i32, month: u8, day: u8, calendar: Calendar) -> Result<Self> {
+        Self::new_with_overflow(year, month, day, calendar, ArithmeticOverflow::Reject)
+    }
+}
+```
+
+These three constructors, `new_with_overflow`, `try_new`, and `new`, are
+fairly flexible and provide full coverage of the Temporal specification.
+
+For instance, take the below constructor:
+
+```js
+const plainDate = new Temporal.PlainDate(2025, 6, 9);
+```
+
+Can easily be conveyed in Rust as:
+
+```rust
+use temporal_rs::PlainDate;
+let plain_date = PlainDate::try_new(2025, 6, 9, Calendar::default())?;
+```
+
+We actually learn some interesting things immediately about the
+JavaScript API from looking at the `temporal_rs` API:
+
+1. `Temporal.PlainDate` constructor can throw.
+2. When the calendar is omitted, the default calendar is used (this will
+   default to the `iso8601` calendar)
+
+Of course, if you somewhat prefer the brevity of the JavaScript API and
+don't want to list the default `Calendar`, `temporal_rs` provides the
+additional constructors `new_iso` and `try_new_iso`.
+
+```rust
+use temporal_rs::PlainDate;
+let plain_date = PlainDate::try_new_iso(2025, 6, 9)?;
+```
+
+Interestingly enough, the `_iso` constructors are actually extensions of
+Temporal specification to provide a similar API in Rust. This is because
+the `_iso` constructors are assumed to exist due to resolving an
+`undefined` calendar to the default ISO calendar.
```

Comment:
I think "extensions of Temporal specification" implies that these are in the way of being implemented in the Temporal spec. Maybe we should remove this paragraph; most of this text already implies that these are just Rust-exclusive APIs.